build_runner/lib/src/server/server.dart

```diff
@@ -110,6 +123,68 @@ class ServeHandler implements BuildState {
   }
 }
 
+/// Class that manages web socket connection handler to inform clients about
+/// build updates
+class BuildUpdatesWebSocketHandler {
+  final activeConnections = <WebSocketChannel>[];
+  final shelf.Handler Function(Function, {Iterable<String> protocols})
+      _handlerFactory;
+  shelf.Handler _internalHandler;
+
+  BuildUpdatesWebSocketHandler([this._handlerFactory = webSocketHandler]) {
+    var untypedTearOff = (webSocket, protocol) =>
```

Comment:
one last nit, can you document why we need to do this, and ideally point at an issue in the original package?

Reply:
Created dart-lang/shelf_web_socket#11